src/screens/ProposalDetail.js

```diff
@@ -379,10 +379,28 @@ function ProposalDetail({
       >
         <SupportProposalScreens
           id={id}
-          onDone={() => setMiltiModalVisible(false)}
           onStakeToProposal={actions.stakeToProposal}
         />
       </MultiModal>
+      <MultiModal
+        visible={modalVisible}
+        onClose={() => setModalVisible(false)}
+        oonClosed={() => setModalMode(null)}
+      >
+        {modalMode === 'challenge' && (
+          <ChallengeProposalScreens
+            agreementActions={{
+              challengeAction: actions.challengeAction,
+              getAllowance: actions.getAllowance,
+              approveChallengeTokenAmount: actions.approveChallengeTokenAmount,
+            }}
+            proposal={proposal}
+          />
+        )}
+        {modalMode === 'settle' && (
+          <SettleProposalScreens proposal={proposal} />
+        )}
```

Comment:
and i think that you need to add the `<SupportProposal>` screens here to call the modal with mode `stake` or something like that